### PR TITLE
feat: [SRM-13463]: Template fix for common assign component

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

Fix for Template scenarios in Assign component

#### Screenshots

https://user-images.githubusercontent.com/81554936/211461263-e5816191-ccd9-45a4-afab-3c94c9f3a061.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx [hunk 1]
```diff
@@ -153,11 +153,12 @@ export default function CommonCustomMetricFormContainer(props: CommonCustomMetri
 
       {/* ⭐️ Assign section */}
       {healthSourceConfig.customMetrics?.assign?.enabled && (
-        <SelectHealthSourceServices
+        <AssignQuery
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx:156]: Please destructure properties from `values`

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx [hunk 2]
```diff
@@ -24,9 +25,32 @@ export default function ServiceInstance({
   continuousVerificationEnabled
 }: ServiceInstanceProps): JSX.Element {
   const { getString } = useStrings()
+  let value = serviceInstance
   const { isQueryRuntimeOrExpression } = useCommonHealthSource()
   const { isTemplate, expressions, sourceData } = useContext(SetupSourceTabsContext)
   const isConnectorRuntimeOrExpression = getIsConnectorRuntimeOrExpression(sourceData.connectorRef)
+  const isServiceInstanceFixed = getMultiTypeFromValue(serviceInstance) === MultiTypeInputType.FIXED
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:32]: Why we need to add separate code for serviceInstance. It should be controlled by formik directly once user the sets the value in the text field.

Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:32]: This is to update value to empty when user switches from fixed value to expression

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx:157]: done

**Comment by praneshg239** [comment]
@deepesh-ui From the video, if we are switching the type of input which had a default value, to runtime or expression and switching back to fixed, it should take the default value again.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/AssignQuery.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ import type { SelectHealthSourceServicesProps } from './AssignQuery.types'
 import AssignSection from './components/AssignSection/AssignSection'
 import css from './AssignQuery.module.scss'
 
-export default function SelectHealthSourceServices({
+export default function AssignQuery({
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/AssignQuery.tsx:15]: 👏

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx [hunk 4]
```diff
@@ -41,13 +65,15 @@ export default function ServiceInstance({
               <FormInput.MultiTextInput
                 label={<ServiceInstanceLabel />}
                 name={CustomMetricFormFieldNames.SERVICE_INSTANCE}
+                onChange={(_value, _valueType, multiType) => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:68]: NIT: Why we are using `_` as prefix for the parameters here? We can still give precise name for them right?

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx [hunk 5]
```diff
@@ -24,9 +25,32 @@ export default function ServiceInstance({
...
+
+  const [metricInstanceMultiType, setMetricPathMultiType] = useState<MultiTypeInputType>(() =>
+    getMultiTypeFromValue(serviceInstance)
+  )
+
+  if (isTemplate) {
+    /**
+     * Update value to empty when user switches from fixed value to expression
+     */
+    const hasFixedValueInExpression =
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:42]: instead doing this, we can simply set the value upon type change right?

**Review by praneshg239 — CHANGES_REQUESTED** [review]
Please check

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx (hunk 4 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:68]: was showing lint error

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx (hunk 5 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/Assign/components/RiskProfile/components/ServiceInstance/ServiceInstance.tsx:42]: done

**Review by harshilbits — APPROVED** [review]

**Pull request merged**